src/dashboard/src/components/administration/views.py

```diff
@@ -223,6 +229,160 @@ def storage(request):
+            # If a location, determine path
+            dir_spec['path'] = _usage_location_path(dir_spec['location_purpose'])
+        elif 'contained_by' in dir_spec:
+            # If contained, make path absolute
+            space = dir_spec['contained_by']
+            absolute_path = os.path.join(dirs[space]['path'], dir_spec['path'])
+            dir_spec['path'] = absolute_path
+
+    # Calculate usage/size, if requested
+    if calculate_usage:
+        for _, dir_spec in dirs.iteritems():
+            if 'contained_by' in dir_spec:
+                # Get size from containing space and calculate usage of specific path
+                space = dir_spec['contained_by']
+                dir_spec['size'] = dirs[space]['size']
+                dir_spec['used'] = _usage_get_directory_used_bytes(dir_spec['path'])
+            else:
+                # Get size/usage of space
+                space_path = dir_spec['path']
+                dir_spec['size'] = _usage_check_directory_volume_size(space_path)
+                dir_spec['used'] = _usage_get_directory_used_bytes(space_path)
+
+    return dirs
+
+def _usage_check_directory_volume_size(path):
+    # Get volume size (in 512 byte blocks)
+    try:
+        output = subprocess.check_output(["df", path])
+
+        # Second line returns disk usage-related values
+        usage_summary = output.split("\n")[1]
+
+        # Split value by whitespace and size (in blocks)
+        size = usage_summary.split()[1]
+
+        return int(size) * 512
+    except Exception, e:
+        logger.exception(str(e))
+        return 0
+
+def _usage_get_directory_used_bytes(path):
+    """ Get total usage in bytes """
+    try:
+        output = subprocess.check_output(["du", "--bytes", "--summarize", path])
+        return output.split("\t")[0]
+    except Exception, e:
+        logger.exception(str(e))
+        return 0
+
+def _usage_location_path(purpose):
+    # Get currently processing location
+    locations = storage_service.get_location(purpose=purpose)
+    location = locations[0]
+    return location['path']
+
+def clear_context(request, dir_id):
+    usage_dirs = _usage_dirs(False)
+    prompt = 'Clear ' + usage_dirs[dir_id]['description'] + '?'
+    cancel_url = reverse("components.administration.views.usage")
+    return RequestContext(request, {'action': 'Delete', 'prompt': prompt, 'cancel_url': cancel_url})
+
+@user_passes_test(lambda u: u.is_superuser, login_url='/forbidden/')
+@decorators.confirm_required('simple_confirm.html', clear_context)
+def usage_clear(request, dir_id):
+    if request.method == 'POST':
+        usage_dirs = _usage_dirs(False)
+        dir_info = usage_dirs[dir_id]
+
+        # Prevent shared directory from being cleared
+        if dir_id == 'shared' or not dir_info:
+            raise Http404
+
+        # Determine if specific subdirectories need to be cleared, rather than
+        # whole directory
+        if 'subdirectories' in dir_info:
```

Comment:
This functionality was originally added to accommodate Transfer Backlog clearing where we needed to clear the contents of two subdirectories, rather than the whole directory's contents. I'm wondering if I should leave this in in case it's useful or if I should remove it?